preprocessing/nextclade/tests/test.py

```diff
@@ -1,348 +1,354 @@
-            expected_output=factory_custom.create_processed_entry(
-                metadata_dict={
-                    "name_required": "name",
-                    "required_collection_date": "2022-11-01",
-                    "concatenated_string": "LOC_5.1/2022-11-01",
-                },
-                metadata_errors=[
-                    (
-                        "sequenced_timestamp",
-                        "Timestamp is  2022-11-01Europe which is not in parseable YYYY-MM-DD. Parsing error: Unknown string format:  2022-11-01Europe",
-                    ),
-                ],
-            ),
-        ),
-        TestCase(
-            name="date_only_year",
-            input=UnprocessedEntryFactory.create_unprocessed_entry(
-                metadata_dict={
-                    "submissionId": "date_only_year",
-                    "collection_date": "2023",
-                    "name_required": "name",
-                    "required_collection_date": "2022-11-01",
-                }
-            ),
-            expected_output=factory_custom.create_processed_entry(
-                metadata_dict={
-                    "collection_date": "2023-01-01",
-                    "name_required": "name",
-                    "required_collection_date": "2022-11-01",
-                    "concatenated_string": "LOC_6.1/2022-11-01",
-                },
-                metadata_errors=[],
-                metadata_warnings=[
-                    (
-                        "collection_date",
-                        "Metadata field collection_date:'2023' - Month and day are missing. Assuming January 1st.",
-                    ),
-                ],
-            ),
-        ),
-        TestCase(
-            name="date_no_day",
-            input=UnprocessedEntryFactory.create_unprocessed_entry(
-                metadata_dict={
-                    "submissionId": "date_no_day",
-                    "collection_date": "2023-12",
-                    "name_required": "name",
-                    "required_collection_date": "2022-11-01",
-                }
-            ),
-            expected_output=factory_custom.create_processed_entry(
-                metadata_dict={
-                    "collection_date": "2023-12-01",
-                    "name_required": "name",
-                    "required_collection_date": "2022-11-01",
-                    "concatenated_string": "LOC_7.1/2022-11-01",
-                },
-                metadata_errors=[],
-                metadata_warnings=[
-                    (
-                        "collection_date",
-                        "Metadata field collection_date:'2023-12' - Day is missing. Assuming the 1st.",
-                    ),
-                ],
+@dataclass
+class Case:
+    name: str
+    metadata: dict[str, str]
+    expected_metadata: dict[str, str]
+    expected_errors: list[tuple[str, str]]
+    expected_warnings: list[tuple[str, str]] = None
+    accession: str = "000999"
```

Comment:
And then here we could call this accession_id to make it clearer that it doesn't yet contain the LOC_ prefix.

Reply:
added